### PR TITLE
Fix load_model for multiple output metrics in dictionary

### PR DESCRIPTION
**Pull request by dieuwkehupkes**
Fixes the problem described in #3958 by adding an extra check in deserialized that recreates original dictionary.

### CODE REVIEW

**Comment by isaacgerg** [comment]
Is this realted to #4143

File: keras/models.py [hunk 1]
```diff
@@ -108,7 +108,15 @@ def deserialize(obj):
         if type(obj) is dict:
             deserialized = {}
             for key, value in obj.items():
-                if value in custom_objects:
+                deserialized[key] = []
+                if type(value) is list:
+                    for element in value:
+                        if element in custom_objects:
+                            deserialized[key].append(custom_objects[value])
```
Comment (fchollet) [inline keras/models.py]: Should be `element` here...

**Comment by fchollet** [comment]
Closing outdated PR. If you still care about the content of the PR, please submit a new PR to `master`, updated for the Keras 2.0 API.

**Comment by dieuwkehupkes** [comment]
Dear Francois,

I submitted a new PR, the solution is still identical to the previous, but
updated for the Keras 2.0 API.

Best,

Dieuwke

2017-03-15 22:39 GMT+01:00 François Chollet <notifications@github.com>:

**Pull request closed without merging**